### PR TITLE
Update address to pubkey, allow more flexibility in request

### PR DESCRIPTION
**Pull request by CriesofCarrots**
This renames address to pubkey in various places. It also borrows from the Stellar ledger app to simplify getPublicKey and getPrivateKey methods and rework the expected dataBuffer to allow for a second level of key derivation (enabling 44'/501'/X'/Y' in addition to 44'/501'/X')

This new code expects 44'/501' as part of the provided dataBuffer; technically, this is not necessary, and I happy to discuss ripping out those 8 bytes that have to be sent from the client application (`derivationPath` in the python test). However a number of established Ledger apps that I happened upon while testing (Ethereum, Stellar, etc) use this format, and I found it very useful to have this consistency when working on client code.

The UX needs more work to display a base58 pubkey on the pubkey confirm screen(s).

### CODE REVIEW

File: src/getPubkey.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#include "getPubkey.h"
+#include "os.h"
+#include "ux.h"
+#include "utils.h"
+
+static uint8_t publicKey[FULL_PUBKEY_LENGTH];
+
+static int read_bip32(const uint8_t *dataBuffer, size_t size, uint32_t *bip32) {
```
Comment (garious) [inline src/getPubkey.c]: This either needs a new name or better docs.  BIP32 is a standard. What's a `uinit32_t *bip32`?

**Comment by CriesofCarrots** [comment]
@garious , fyi I have no merge power here

**Comment by garious** [comment]
you now have merge power

**Comment by CriesofCarrots** [comment]
image

**Pull request merged**